### PR TITLE
DOMA-5149 change constraint in model and add deletedAt to gql for UserExternalIdentity

### PR DESCRIPTION
**Pull request by VKislov**
Changed the `UserExternalIdentity` `constraint` so that the check does not consider deleted entries. Added `deletedAt` to `gql` so that it would be possible to filter the search for entities. Added cleaning the entities associated with the user who is being reset via `ResetUserService`. 
Also fixed the problem of the infinite rerender with a 500 error when trying to go behind a deleted user.

### CODE REVIEW

File: apps/condo/domains/user/schema/ResetUserService.js [hunk 1]
```diff
@@ -109,6 +110,16 @@ const ResetUserService = new GQLCustomSchema('ResetUserService', {
                     await OrganizationEmployee.softDelete(context, employee.id, { dv: 1, sender })
                 }
 
+                const accordingUserExternalIdentity = await UserExternalIdentity.getAll(context, {
```
Comment (pahaz) [inline apps/condo/domains/user/schema/ResetUserService.js:113]: You need to add a test for that

File: apps/condo/domains/organization/integrations/sbbol/sync/syncUser.js [hunk 2]
```diff
@@ -74,6 +74,7 @@ const syncUser = async ({ context: { context, keystone }, userInfo, identityId }
     const identityWhereStatement = {
         identityId,
         identityType: SBBOL_IDP_TYPE,
+        deletedAt: null,
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncUser.js:77]: You don't need to use deleted at?

File: apps/condo/domains/common/components/UseDeskWidget.tsx [hunk 3]
```diff
@@ -28,24 +28,29 @@ const UseDeskWidget: React.FC = () => {
...
-                            id: useDeskFieldsIdsMap.role, value: get(link, ['role', 'name'], null),
-                        },
-                    ],
-            })
+        if (UseDeskWidgetId && isFunction(userIdentify)) {
+            const name = get(link, 'name')
+            const email = get(user, 'email')
+            const phone = get(user, 'phone')
+            userIdentify(
+                {
```
Comment (pahaz) [inline apps/condo/domains/common/components/UseDeskWidget.tsx:36]:
```suggestion
            userIdentify({
```

File: apps/condo/domains/organization/integrations/sbbol/sync/syncUser.js (hunk 2 above)
Comment (VKislov) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncUser.js:77]: Screenshot 2023-02-01 at 12 50 12

Comment (VKislov) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncUser.js:77]: without it I get a deleted instance

Comment (pahaz) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncUser.js:77]: Looks like a bug! You should not use it

**Review by sitozzz — APPROVED** [review]

**Pull request merged**